carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -1754,17 +1756,36 @@ public By generateByForList(By by, int index) {
 
     // old functionality to remove completely after successfull testing
     private ExpectedCondition<?> getDefaultCondition(By myBy) {
-        // generate the most popular wiatCondition to check if element visible or present
+        // generate the most popular waitCondition to check if element visible or present
         ExpectedCondition<?> waitCondition = null;
-        if (element != null) {
-            waitCondition = ExpectedConditions.or(ExpectedConditions.presenceOfElementLocated(myBy),
-                    ExpectedConditions.visibilityOfElementLocated(myBy),
-                    ExpectedConditions.visibilityOf(element));
-        } else {
-            waitCondition = ExpectedConditions.or(ExpectedConditions.presenceOfElementLocated(myBy),
-                    ExpectedConditions.visibilityOfElementLocated(myBy));
+        switch (loadingStrategy) {
+        case BY_PRESENCE: {
+            if (element != null) {
+                waitCondition = ExpectedConditions.or(ExpectedConditions.presenceOfElementLocated(myBy), ExpectedConditions.visibilityOf(element));
+            } else {
+                waitCondition = ExpectedConditions.presenceOfElementLocated(myBy);
+            }
+            break;
+        }
+        case BY_VISIBILITY: {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ci.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS1199)